Source/TimeIntegration/StrainRate.cpp

```diff
@@ -136,3 +134,93 @@ ComputeStrainRate(const int &i, const int &j, const int &k,
 
   return strainRate;
 }
+
+AMREX_GPU_DEVICE
+Real
+ComputeExpansionRate(const int &i, const int &j, const int &k,
+                     const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                     const enum NextOrPrev &nextOrPrev,
+                     const enum MomentumEqn &momentumEqn,
+                     const enum DiffusionDir &diffDir,
+                     const GpuArray<Real, AMREX_SPACEDIM>& cellSize) {
+    Real dx = cellSize[0];
```

Comment:
Another potential performance optimization would be to pass invCellSize here instead of cellSize to avoid all the divides (and similarly in other parts of the code).